config.py

```
@@ -63,0 +1,272 @@
+SECRET_KEY_SUBSTRINGS = [
+    "password", "token", "secret", "passwd", 
+    "smtp_user", "smtp_pass", "api_key" # Added common term
+]
+
+class Config:
+    def __init__(self):
+        self._config = {}  # Loaded from YAML
+        self._loaded = False
+        self._env_file_loaded = False # Tracks if a .env file was processed
+
+    def _is_secret(self, key_name_parts):
```

Ruff (0.11.9)

Missing return type annotation for private function `_is_secret`

Add return type annotation: `bool`

(ANN202)